libraries/common/src/main/java/androidx/media3/common/Format.java

```diff
@@ -1190,6 +1190,9 @@ public Format withManifestFormatInfo(Format manifestFormat) {
         codecs = codecsOfType;
       }
     }
+    if (MimeTypes.VIDEO_DOLBY_VISION.equals(sampleMimeType)) {
```

Comment:
The purpose of this special check is to update codecs with manifest codec info. For my test stream,
manifestFormat.codecs = "dvhe.08.01" while codecs = "hev1.08.01" before this special handling. (So that L1185 is false and corresponding logic is not touched.)

Reply:
Interesting. Does this happen because the extractor doesn't know this is a DV format? For MP4 containers for example I'd assume they have a `dvcC` or `dvvC`, which should results in the `format.codecs` already populated with the expected values.

Reply:
During the fmp4 paring, the parser finds the 'dvcc' or 'dvvc' box [here](https://github.com/androidx/media/blob/1.5.0/libraries/extractor/src/main/java/androidx/media3/extractor/mp4/BoxParser.java#L1337), and corresponding codec string is updated in [DolbyVisionConfig::parse()](https://github.com/androidx/media/blob/1.5.0/libraries/extractor/src/main/java/androidx/media3/extractor/DolbyVisionConfig.java#L44). After tracking the history of this file, I finally found [this one](https://github.com/google/ExoPlayer/pull/5438). The author was Dolby engineer but is Google engineer now. :)

I'm not sure whether there is any side effect on codec selection if I change the logic (e.g. from hev1 to dvhe). My understanding is OK since there is [a method](https://github.com/androidx/media/blob/1.5.0/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/mediacodec/MediaCodecUtil.java#L322) to handle the case (A non Dolby Vision licensed device to play Dolby Vision content.) But I didn't do a full testing.

Do you want to change the logic in [DolbyVisionConfig::parse()](https://github.com/androidx/media/blob/1.5.0/libraries/extractor/src/main/java/androidx/media3/extractor/DolbyVisionConfig.java#L44)? or using a new pull request to do that? At least, I need to provide a new PR for profile 10 handling in that method.

Reply:
Can we assume the dvcc' or 'dvvc' box is always there and we should always be able to tell from the actual media container which Dolby Vision codec we have? If so, I think it's better to fix the logic in [DolbyVisionConfig::parse()](https://github.com/androidx/media/blob/1.5.0/libraries/extractor/src/main/java/androidx/media3/extractor/DolbyVisionConfig.java#L44) and not make the change in `Format.withManifestFormatInfo`. 

The reason for that is that `withManifestFormatInfo` merges information from the actual media files and the manifest. In some cases like labels and language it's generally better to rely on the manifest. In other cases like codecs, it's usually better to rely on the actual media file as it's more likely to get it right (or has more details) than the manifest in case there are any differences. So if we can be certain that the  dvcc' or 'dvvc' boxes always exist for a Dolby Vision stream (and we fix our util method to set the right codec strings for all profiles), then it seems better to rely on this logic. 

> or using a new pull request to do that?

That seems like a separate change and I think it's easier if I continue to merge the current pull request as it is and we can look at the other change separately?

Reply:
> Can we assume the dvcc' or 'dvvc' box is always there and we should always be able to tell from the actual media container which Dolby Vision codec we have? If so, I think it's better to fix the logic in [DolbyVisionConfig::parse()](https://github.com/androidx/media/blob/1.5.0/libraries/extractor/src/main/java/androidx/media3/extractor/DolbyVisionConfig.java#L44) and not make the change in `Format.withManifestFormatInfo`.

Agree. I think there is always a Dolby Vision box (dvcc or dvvc) to indicate the codec info. I need to double confirm it.

> That seems like a separate change and I think it's easier if I continue to merge the current pull request as it is and we can look at the other change separately?

I'll submit a new pull request after this one is merged. Is it fine for you?

Reply:
I checked the lasted spec today.

_• For profiles less than or equal to 7: dvcC
• For profiles 8 to 10: dvvC
• For profile 11 to 19: dvwC
• For profile 20: dvcC
• For profiles greater than 20: dvwC_

Currently, we just need to support profile 4/5/7/8/9/10 and potentially support profile 20 so that there is always a 'dvcC' or 'dvvC' box exist.